hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/fair/FSQueue.java

```diff
@@ -158,26 +158,37 @@ public Resource getMinShare() {
     return minShare;
   }
 
-  public void setMaxShare(Resource maxShare){
+  public void setMaxShare(ConfigurableResource maxShare){
     this.maxShare = maxShare;
   }
 
+  @Override
+  public Resource getMaxShare() {
+    Resource maxResource = maxShare.getResource(scheduler.getClusterResource());
+
+    // Set max resource to min resource if min resource is greater than max
+    // resource
+    if(Resources.greaterThan(scheduler.getResourceCalculator(),
+        scheduler.getClusterResource(), minShare, maxResource)) {
```

Comment:
This should likely by componentWiseMax. See Daniel's patch on YARN-6964.

Reply:
Yeah, you should avoid greaterThan() (et al) if you can.